jujugui/static/gui/src/app/components/header-breadcrumb/test-header-breadcrumb.js

```diff
@@ -169,4 +172,42 @@ describe('HeaderBreadcrumb', () => {
     assert.equal(output.props.children[1], undefined);
   });
 
+  it('triggers a state change when profile link is clicked', () => {
+    var app = {app:'app'};
+    var env = {env: 'env'};
+    var envName = 'bar';
+    var jem = {jem: 'jem'};
+    var envList = ['envList'];
+    var changeState = sinon.stub();
+    var authDetails = {
+      user: 'foo',
+      usernameDisplay: 'Foo'
+    };
+    var component = jsTestUtils.shallowRender(
+      <juju.components.HeaderBreadcrumb
+        app={app}
+        env={env}
+        envName={envName}
+        dbEnvironmentSet={sinon.stub()}
+        jem={jem}
+        envList={envList}
+        changeState={changeState}
+        getAppState={sinon.stub()}
+        showConnectingMask={sinon.stub()}
+        authDetails={authDetails}
+        showEnvSwitcher={true}
+        switchModel={sinon.stub()}
+        uncommittedChanges={false} />, true);
+    var instance = component.getMountedInstance();
+    var output = component.getRenderOutput();
+    instance._handleProfileClick({
+      preventDefault: sinon.stub()
+    });
+    assert.equal(changeState.called, true,
+                 'changeState was not called');
+    var state = changeState.args[0][0];
+    assert.deepEqual(state.sectionC.component, 'profile',
```

Comment:
Yeah, whoops, my bad. I originally compared the entire state object but then decided that the component was the primary thing we wanted to assert on. That way if we add other information to the state object in the future the test doesn't necessarily break.